conan_provider.cmake

```diff
@@ -470,7 +470,7 @@ macro(conan_set_program_path)
         string(JSON _dep_binary GET ${conan_stdout} graph nodes ${_node_number} binary)
         string(JSON _dep_context GET ${conan_stdout} graph nodes ${_node_number} context)
         
-        if(NOT _dep_binary STREQUAL "Cache" OR NOT _dep_context STREQUAL "build")
+        if(_dep_binary STREQUAL "Skip" OR NOT _dep_context STREQUAL "build")
```

Comment:
Wouldn't you play safer doing a comparison against a lowercase value?:

```
if(_dep_binary_lowercase STREQUAL "skip" OR NOT _dep_context_lowercase STREQUAL "build")
```

Reply:
Or even defining those `"Skip"`, `"build"`... somewhere as constants. Although that may be overkill. 😄 This type of code always looks to me like a bit like "magic numbers".

Reply:
this part of Conan's json formatter, so it's essentially a public API - if this ever changed in Conan, our tests (in Conan) would fail straight away so we would know that would be a breaking change (minimising the chance of this ever happening - so not sure we can something by doing lowercase).

they could be constants but then we would have more indirection given how CMake expands variables but I'll consider that